### ISSUE TITLE
`Oceananigans.BoundaryConditions.ContinuousBoundaryFunction` warnings in examples

### ISSUE THREAD

**Issue by navidcy** [CliMA/Oceananigans.jl#1775]
This warning appears in the log when we run the simulation:

```Julia
┌ Warning: some parameters could not be resolved for type Oceananigans.BoundaryConditions.ContinuousBoundaryFunction{Center,Center,Nothing,24,Main.__atexample__named__ocean_wind_mixing_and_convection.#Qˢ,Float64,Tuple{Symbol},Tuple{Int64},Tuple{typeof(Oceananigans.Operators.identity4)}}; reconstructing
└ @ JLD2 ~/.julia/packages/JLD2/1i6ZM/src/data/reconstructing_datatypes.jl:440
```

See, e.g., https://clima.github.io/OceananigansDocumentation/stable/generated/ocean_wind_mixing_and_convection/

Is this something we need to address or make the example better so that it does not appear? Users might be intimidated...

**Comment by glwagner** [comment]
I can't figure out why that warning is spit at every time step.

I'd note that

https://github.com/CliMA/Oceananigans.jl/blob/87270dea8d1553339bb8f806034df462604111d2/src/OutputWriters/jld2_output_writer.jl#L189

was added in #1641, which won't work in some cases (eg when a boundary condition contains a `CuArray`).

But I don't see anything in

https://github.com/CliMA/Oceananigans.jl/blob/87270dea8d1553339bb8f806034df462604111d2/src/OutputWriters/jld2_output_writer.jl#L201-L232

or

https://github.com/CliMA/Oceananigans.jl/blob/87270dea8d1553339bb8f806034df462604111d2/src/OutputWriters/jld2_output_writer.jl#L242-L250

that would throw such a warning. And the above are the parts that are called whenever output is asked for.

**Comment by navidcy** [comment]
This was resolved with #1847 and in particular the modifications by @glwagner in the output writer.
